src/lesson_7.ts

```diff
@@ -0,0 +1,297 @@
+enum EmployeeStatus {
+  Active,
+  Inactive,
+  OnUnpaidLeave,
+}
+
+enum EmployeeType {
+  PreHiredEmployee,
+  Employee,
+}
+
+class PreHiredEmployee {
+  private firstName: string;
+  private lastName: string;
+  private salary: number;
+  private bankAccountNumber: string;
+  // Discriminant type
+  discriminantType: EmployeeType = EmployeeType.PreHiredEmployee;
+
+  constructor(
+    firstName: string,
+    lastName: string,
+    salary: number,
+    bankAccountNumber: string
+  ) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.salary = salary;
+    this.bankAccountNumber = bankAccountNumber;
+  }
+  getFirstName(): string {
+    return this.firstName;
+  }
+  getLastName(): string {
+    return this.lastName;
+  }
+  getSalary(): number {
+    return this.salary;
+  }
+  getBankAccountNumber(): string {
+    return this.bankAccountNumber;
+  }
+  getDiscriminantType(): EmployeeType {
+    return this.discriminantType;
+  }
+  setFirstName(firstName: string): void {
+    this.firstName = firstName;
+  }
+  setLastName(lastName: string): void {
+    this.lastName = lastName;
+  }
+  setSalary(salary: number): void {
+    this.salary = salary;
+  }
+  setBankAccountNumber(bankAccountNumber: string): void {
```

Comment:
Could you please explain why you are not using get/set accessors instead of having all these methods?

Reply:
i did not work with classes on ts a lot; using methods is just more familiar